includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -284,61 +266,64 @@ public function should_hide_widgets( bool $return ): bool {
-	 */
-	private function cart_contains_renewal() {
-		if ( ! function_exists( 'wcs_cart_contains_renewal' ) ) {
-			return false;
-		}
-		return wcs_cart_contains_renewal();
-	}
-
-	/**
-	 * Checks an order  to see if it contains a subscription product renewal.
+	 * This checks both the cart itself and the session. This is due to there are times when an item may be present in
+	 * one place and not the other. We need to make sure that if an item is in either we are not creating double conversions.
 	 *
-	 * @param object $order Order object.
+	 * @param string $type The type of subscription to look for in the cart.
 	 *
-	 * @return bool The cart item containing the renewal as an array, else false.
+	 * @return mixed False if none found, or the subscription cart item as an array.
 	 */
-	private function order_contains_renewal( $order ): bool {
-		if ( ! function_exists( 'wcs_order_contains_renewal' ) ) {
+	private function get_subscription_type_from_cart( $type ) {
+		// Make sure we're looking for allowed types.
+		if ( ! in_array( $type, self::SUBSCRIPTION_TYPES, true ) ) {
 			return false;
 		}
-		return wcs_order_contains_renewal( $order );
-	}
 
-	/**
-	 * Gets the subscription switch items out of the cart.
-	 *
-	 * @return array Empty array or the cart items in an array..
-	 */
-	private function get_subscription_switch_cart_items(): array {
-		if ( ! function_exists( 'wcs_get_order_type_cart_items' ) ) {
-			return [];
+		// Set the sub type cart key.
+		$subscription_type = 'subscription_' . $type;
+
+		// Go through each cart item and if it matches the type, return that item.
+		if ( isset( WC()->cart ) && is_array( WC()->cart->cart_contents ) && ! empty( WC()->cart->cart_contents ) ) {
+			foreach ( WC()->cart->cart_contents as $cart_item ) {
+				if ( isset( $cart_item[ $subscription_type ] ) ) {
+					return $cart_item;
+				}
+			}
+		}
+
+		// Go through each session cart item and if it matches the type, return that item.
+		if ( isset( WC()->session ) && is_array( WC()->session->get( 'cart' ) ) && ! empty( WC()->session->get( 'cart' ) ) ) {
+			foreach ( WC()->session->get( 'cart' ) as $cart_item ) {
+				if ( isset( $cart_item[ $subscription_type ] ) ) {
+					return $cart_item;
+				}
+			}
 		}
-		return wcs_get_order_type_cart_items( 'switch' );
+
+		return false;
 	}
 
 	/**
 	 * Getter for subscription objects.
 	 *
 	 * @param  mixed $the_subscription Post object or post ID of the order.
+	 *
 	 * @return mixed The subscription object, or false if it cannot be found.
-	 *               Note: this is WC_Subscription|bool in normal use, but in tests
-	 *               we use WC_Order to simulate a subscription (hence `mixed`).
+	 *               Note: This should be WC_Subscription|bool, but Psalm throws errors like:
+	 *                     Docblock-defined class, interface or enum named WC_Subscription does not exist (see https://psalm.dev/200)
```

Comment:
If there are any thoughts on how to fix this, I am open to it. It may be better to leave it like this since Subs is going to be removed from WCPay.

Reply:
I would say it may be best to leave it since subscriptions will be removed from WCPay.